tests/models/test_taskinstance.py

```diff
@@ -24,26 +24,39 @@
 
 import pendulum
 from freezegun import freeze_time
-from unittest.mock import patch, mock_open
+from mock import mock_open, patch
 from parameterized import parameterized
 
-from airflow import models, settings, configuration
+from airflow import configuration, models, settings
 from airflow.contrib.sensors.python_sensor import PythonSensor
 from airflow.exceptions import AirflowException, AirflowSkipException
-from airflow.models import DAG, TaskFail, TaskInstance as TI, TaskReschedule, DagRun
+from airflow.models import DAG, DagRun, Pool, TaskFail, TaskInstance as TI, TaskReschedule
 from airflow.operators.bash_operator import BashOperator
 from airflow.operators.dummy_operator import DummyOperator
 from airflow.operators.python_operator import PythonOperator
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.ti_deps.dep_context import REQUEUEABLE_DEPS, RUNNABLE_STATES, RUNNING_DEPS
+from airflow.ti_deps.deps.base_ti_dep import TIDepStatus
 from airflow.ti_deps.deps.trigger_rule_dep import TriggerRuleDep
 from airflow.utils import timezone
 from airflow.utils.db import create_session
 from airflow.utils.state import State
 from tests.models import DEFAULT_DATE
+from tests.test_utils import db
 
 
 class TaskInstanceTest(unittest.TestCase):
 
+    def setUp(self):
+        db.clear_db_pools()
+        session = settings.Session()
```

Comment:
Please use `with create_session() as session:` here as well.

Reply:
roger that.